x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -1207,10 +1214,25 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
       ...packagePolicyUpdates.flatMap((p) => p.policy_ids),
       ...oldPackagePolicies.flatMap((p) => p.policy_ids),
     ]);
-    logger.debug(`Bumping revision of associated agent policies ${associatedPolicyIds}`);
+
+    const [endpointPackagePolicyUpdatesIds, endpointOldPackagePoliciesIds] = [
+      packagePolicyUpdates,
+      oldPackagePolicies,
+    ].map((packagePolicies) =>
```

Comment:
Sets instead of arrays could be built here, so the checks in lines 1231 and 1232 will have constant time complexity

Reply:
`const [endpointPackagePolicyUpdatesIds, endpointOldPackagePoliciesIds] = [
  packagePolicyUpdates,
  oldPackagePolicies,
].map((packagePolicies) =>
  new Set(
    packagePolicies
      .filter((p) => p.package?.name === 'endpoint')
      .map((p) => p.policy_ids)
      .flat()
  )
);` Like this, right?

Reply:
May I also add that instead of iterating twice using filter and map you could use reduce to iterate only once. 😅